handler/interface.go

```diff
@@ -5,16 +5,21 @@ import (
 
 	"github.com/ONSdigital/dp-api-clients-go/cantabular"
 	"github.com/ONSdigital/dp-api-clients-go/recipe"
+	"github.com/ONSdigital/dp-api-clients-go/dataset"
 )
 
 type cantabularClient interface{
 	GetCodebook(context.Context, cantabular.GetCodebookRequest) (*cantabular.GetCodebookResponse, error)
 }
 
 type datasetAPIClient interface{
-	// Not sure which calls we will be making yet
+	PutInstance(context.Context, string, string, string, string, dataset.UpdateInstance) error
 }
 
 type recipeAPIClient interface{
 	GetRecipe(context.Context, string, string, string) (*recipe.Recipe, error)
+}
+
+type kafkaProducer interface{
```

Comment:
We can delete this interface, there is an interface and mock package in dp-kafka (interface and mock with stub functionality)
- https://github.com/ONSdigital/dp-kafka/blob/main/producer.go#L17
- https://github.com/ONSdigital/dp-kafka/blob/main/kafkatest/message_producer.go